src/runtime/virtcontainers/qemu.go

```diff
@@ -2568,15 +2580,8 @@ func (q *qemu) Cleanup(ctx context.Context) error {
 }
 
 func (q *qemu) GetPids() []int {
-	data, err := os.ReadFile(q.qemuConfig.PidFile)
-	if err != nil {
-		q.Logger().WithError(err).Error("Could not read qemu pid file")
-		return []int{0}
-	}
-
-	pid, err := strconv.Atoi(strings.Trim(string(data), "\n\t "))
-	if err != nil {
-		q.Logger().WithError(err).Error("Could not convert string to int")
+	pid := q.state.Pid
+	if pid == 0 {
 		return []int{0}
```

Comment:
I wonder why we return an array with a zero inside here? This seems to sort of rely on the fact that 0 is an error value of `GetHypervisorPid`, but it's not very clean, and it does not match what `GetHypervisorPid` actually considers the error case:

```
()
	if len(pids) == 0 {
		return 0
	}
	return pids[0]
}
```

Reply:
0 is guaranteed to never be a valid pid : []int{} and []int{0} are thus kind of _equivalent_ . But I agree this looks messy. I'll fix that.